lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp

```diff
@@ -1211,6 +1211,25 @@ struct Int2BVOpLowering : public SMTLoweringPattern<Int2BVOp> {
   }
 };
 
+/// Lower `smt.bv2int` operations to the following Z3 API function call.
+/// ```
+/// Z3_ast Z3_API Z3_mk_bv2int(Z3_context c, Z3_ast t1, bool is_signed)
+/// ```
+struct BV2IntOpLowering : public SMTLoweringPattern<BV2IntOp> {
+  using SMTLoweringPattern::SMTLoweringPattern;
+
+  LogicalResult
+  matchAndRewrite(BV2IntOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    Value isSignedConst = rewriter.create<LLVM::ConstantOp>(
+        op->getLoc(), rewriter.getI1Type(), op.getIsSigned());
```

Comment:
I'm not sure if we can safely use the `i1` type here. I've seen `bool` being compiled to `i8` in LLVM, and I'm not sure if `i32` could also occur.
We should eventually insert a runtime library between this lowering and the Z3 API to have more control over that.
I'd at least insert some TODO or FIXME here.